ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/LocalCommandsFile.java

```diff
@@ -59,7 +59,7 @@ public static LocalCommandsFile createWriteable(final File file) {
     return new LocalCommandsFile(file, true);
   }
 
-  public void write(final LocalCommand localCommand) throws IOException {
+  public synchronized void write(final LocalCommand localCommand) throws IOException {
```

Comment:
Do we want to do this on close as well, to ensure that we don't close it partway through writing a record?

Reply:
Good point.